CONVENTIONS.md

```diff
@@ -129,6 +129,72 @@ does not match the Go definitions that developers use, but it has the advantage
 that generated API documentation and `oc explain` output show the correct field
 names that end users use, and the end-user experience is more important.
 
+### Write User Readable Documentation in Godoc
+
+Godoc text is generated into both Swagger and OpenAPI schemas which are then used
+in tools such as `oc explain` to provide users descriptions of how to use our products.
+
+The Godoc on any field in our API should be sufficiently explained such that an end user
+understands the following:
+* What is the purpose of this field? What does it allow them to achieve?
+* How does setting this field interact with other fields or features?
+* What are the limitations of this field?
+  * Does it have any maximum or minimum value?
+  * If it is a string value, are the values limited to a specific list or can it be free form,
+  must it meet a certain regex?
+  * See the [validation docs](https://book.kubebuilder.io/reference/markers/crd-validation.html) for inspiration on more validations to apply.
+* When optional, what happens when the field is omitted?
+  * You may choose to set a default value within the API or have a controller default the value
+    at runtime
+
+For example:
+
+```go
+// Example enables developers to understand how to write user facing documentation
+// within the Godoc of their API types.
+// Example is used within the wider Conventions to improve the end user experience
+// and is a required convention.
+// At least one value must be provided within the example and the type should be set
+// appropriately.
+// +kubebuilder:validation:Required
+// + ---
+// + Note that this comment line will not end up in the generated API schema as it is
+// + preceded by a `+`. The `---` also prevents anything after it from being added to
+// + the swagger docs.
+// + This can be used to add notes for developers that aren't intended for end users.
+type Example struct {
+  // Type allows the user to determine how to interpret the example given.
+  // It must be set to one of the following values; Documentation, Convention, or Mixed.
```

Comment:
Is it necessary to type out the values for an enum if they're also included in the kubebuilder validation directive?

Reply:
If you don't add them to the godoc, then they won't appear in `oc explain` as it doesn't print out the enum values. So i think the convention has been to include them in the documentation explicitly as well as having the enum for validation purposes